drivers/bridge/setup_ip_tables.go

```diff
@@ -13,14 +13,43 @@ const (
 	DockerChain = "DOCKER"
 )
 
-func (n *bridgeNetwork) setupIPTables(config *networkConfiguration, i *bridgeInterface) error {
+func setupIPChains(config *configuration) (*iptables.ChainInfo, *iptables.ChainInfo, error) {
 	// Sanity check.
 	if config.EnableIPTables == false {
-		return IPTableCfgError(config.BridgeName)
+		return nil, nil, fmt.Errorf("Cannot create new chains, driver EnableIPTable is disabled")
 	}
 
 	hairpinMode := !config.EnableUserlandProxy
 
+	natchain, err := iptables.NewChain(DockerChain, iptables.Nat)
+	if err != nil {
+		return nil, nil, fmt.Errorf("Failed to create NAT chain: %s", err.Error())
+	}
+
+	// TODO(banix): Move the following block; keep this function for just creating the chains
+	natchain, err = iptables.ProgramChain(natchain, hairpinMode)
+	if err != nil {
+		return natchain, nil, fmt.Errorf("Failed to program NAT chain: %s", err.Error())
```

Comment:
Filter configuration is going to fail if either of the two chain creation fails.
I don't think we need to return partial success here. To keep it simple, let's return the chains only if all operations (creation, programming) succeeded.
So in case of failures in this function, I think we should undo whatever succeeded up to this point.